src/chapter-03/src/core.rs

```diff
@@ -25,17 +13,20 @@ pub struct ShortenUrlResponse {
     shortened_url: String,
 }
 
-pub struct UrlShortener {}
+#[derive(Default, Debug)]
```

Comment:
Derived `Default` and `Debug`. I am also using the `default()` method in the constructor. This is a convenient shortcut, since we don't have to manually create an empty `HashMap` and wrap it in a `RwLock`. Should make the code a little bit cleaner and shorter.